packages/web3/src/codec/codec.ts

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+import { Parser } from 'binary-parser'
+
+export interface Codec<T> {
+  parser: Parser
+  encode(input: T): Buffer
```

Comment:
It seems there are some issues with the `Buffer`, it's not compatible with the buffer we are using in `web3` package. We need to import it from `buffer/`: `import { Buffer } from `buffer/`, I have fixed it in the `ralph-map` branch: https://github.com/alephium/alephium-web3/pull/337/commits/4bc2c24aca06cc541c1993852b0105e083ee0d3d